### ISSUE TITLE
Minecraft Bedrock crashes after being on/in water for a while

### ISSUE THREAD

**Issue by xproot** [minecraft-linux/mcpelauncher-manifest#596]
Title explains everything, makes the playerdata unusable (crashes everytime i go back into the world, i don't crash in android after being a while in water but if i try to log in in android while i crashed on mcpelauncher i crash on android)
LOG: https://paste.debian.net/plain/1209664

**Comment by xproot** [comment]
my world just broke because of this and its just sad

**Comment by LoganSimonsen** [comment]
@xproot the same thing has happened to me a couple of times now in the last two days. Here is how I worked around the problem, which is annoying but should work for you.

1. log into the realm or server with a different microsoft account
2. on your android phone or tablet sign into your primary (broken account)
3. on your linux desktop with your secondary account, be ready to teleport your primary account by using command `/teleport <playername> <yourname>` which should teleport your primary account player to your secondary account player.

SUPER annoying, but it did help me at least get unstuck from the situation... unless I got back in water and it crashed again.

**Comment by xproot** [comment]
What about when you don't want to enable cheats in your world (as enabling them disables achievements)

**Comment by xproot** [comment]
Funnily enough while messing around I found out that **THIS BUG CAN CRASH SERVERS**, just need to go to your android device after you have nuked your player data (after crashing on water)

I actually recorded it so here it is, not sure of the exact reproduction methods: https://youtu.be/Sc7ZYFSpmhE

**Comment by LoganSimonsen** [comment]
> What about when you don't want to enable cheats in your world (as enabling them disables achievements)
> Also I was playing on a Local world and copying that to another device would be annoying (If you don't copy it and just sign in as another account you still have your main's account playerdata) Guess the workaround to **that** would be running a server

If it's a local game then I believe there is a way to edit the local file using some tool someone mentioned in another issue thread. Sucks though. 

Crashing servers seems bad for sure, but I didn't seem to have that problem on my realm.

**Comment by xproot** [comment]
> If it's a local game then **I believe there is a way to edit the local file using some tool** someone mentioned in another issue thread. Sucks though.

Oh my gosh do you have that tool? I have been searching the WHOLE internet for one trying to fix my world but there was none!

**Comment by SixPathsSolutions** [comment]
I have lost 3-5 worlds currently due to this bug. Does seem highly correlated to water. Anything we can do to help find a solution?

**Comment by starfox101** [comment]
same here! And no controller! pretty much if I get in water moving fast, it crashes!

**Comment by YiB-PC** [comment]
It crashes when you enter the water and discard items from your inventory.
(translated with deepl)

**Comment by xproot** [comment]
False, I've had a lot of tests which don't include discarding items from my
inventory yet it still crashes.

**Comment by thrhacker** [comment]
I was standing under flowing water when my air bar went to kalf the game froze. I was also putting items into a chest.

**Comment by xproot** [comment]
I want to add that the maintainer of this launcher is very aware of this
bug, yet is probably never going to fix it because it's too complicated.
